frontend/src/app/main-table/cost-table/cost-table.component.ts

```diff
@@ -9,27 +15,22 @@ import { AggregateCostResponse } from 'src/app/services/kubecost.service';
 export class CostTableComponent{
   @Input() aggregatedCost: AggregateCostResponse; 
   @Input() currNamespace: string;
-
-  message: string;
+  
+  AsyncStatus = AsyncStatus
 
   formatCost(value: number): string {
     return '$' + (value > 0 ? Math.max(value, 0.01) : 0).toFixed(2)
   }
 
-  checkStatus(): boolean {
-    if (this.aggregatedCost==null) {
-      this.message = "Waiting for response..."
-      return false;
-    }
-    else {
-      if (Object.keys(this.aggregatedCost).includes('log')) {
-        this.message = "Failed to retrieve cost information."
-        return false;
-      }
-      else {
-        return true;
-      }
-    }
+  getStatus(): AsyncStatus {
+    if (this.currNamespace=='' || this.aggregatedCost==null) {
```

Comment:
I don't think this component should know about the current namespace. That is a concern of the parent component. If the parent wants to reset the state of this table when the namespace changes, then it can set the `aggregatedCost` parameter to `null`.